subprojects/docs/src/docs/userguide/more_about_tasks.adoc

```diff
@@ -831,112 +947,175 @@ Gradle also considers the _code_ of the task as part of the inputs to the task.
+----
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/userguide/tasks/incrementalBuild/incrementalBuildAdvanced/build.gradle[tag=task-arg-method]
+----
+
+.Output of`gradle processTemplates2`
+----
+> gradle processTemplates2
+include::{samplesPath}/userguide/tasks/incrementalBuild/incrementalBuildAdvanced/incrementalBuildCustomMethodsWithTaskArg.out[]
+----
 
 This technique can make your custom task easier to use and result in cleaner build files.
 As an added benefit, our use of `getProject().getLayout().files()` means that our custom method can set up an inferred task dependency.
 
-One last thing to note: if you are developing a task that takes collections of source files as inputs, like this example, consider using the built-in api:org.gradle.api.tasks.SourceTask[]. It will save you having to implement some of the plumbing that we put into `ProcessTemplates`.
+One last thing to note: if you are developing a task that takes collections of source files as inputs, like this example, consider using the built-in link:{groovyDslPath}/org.gradle.api.tasks.SourceTask.html[SourceTask]. It will save you having to implement some of the plumbing that we put into `ProcessTemplates`.
 
 [[sec:link_output_dir_to_input_files]]
-===== Linking an `@OutputDirectory` to an `@InputFiles`
+==== Linking an `@OutputDirectory` to an `@InputFiles`
 
 When you want to link the output of one task to the input of another, the types often match and a simple property assignment will provide that link. For example, a `File` output property can be assigned to a `File` input.
 
 Unfortunately, this approach breaks down when you want the files in a task’s `@OutputDirectory` (of type `File`) to become the source for another task’s `@InputFiles` property (of type `FileCollection`). Since the two have different types, property assignment won’t work.
 
 As an example, imagine you want to use the output of a Java compilation task - via the `destinationDir` property - as the input of a custom task that instruments a set of files containing Java bytecode. This custom task, which we’ll call `Instrument`, has a `classFiles` property annotated with `@InputFiles`. You might initially try to configure the task like so:
 
-++++
-<sample id="incrementalBuildBadInputFilesConfig" dir="userguide/tasks/incrementalBuild/incrementalBuildAdvanced" title="Failed attempt at setting up an inferred task dependency">
-                    <sourcefile file="build.gradle" snippet="failed-inferred-task-dep"/>
-                    <output args="clean badInstrumentClasses" ignoreExtraLines="true" ignoreLineOrder="true"/>
-                </sample>
-++++
+=== Example: Failed attempt at setting up an inferred task dependency
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/userguide/tasks/incrementalBuild/incrementalBuildAdvanced/build.gradle[tag=failed-inferred-task-dep]
+----
+
+.Output of **`gradle clean badInstrumentClasses`**
```

Comment:
I see this in various places.. is this supposed to be bolded?

Reply:
It's bolded in the old docs... "Output of" is now also bolded (as all titles are now). Looks off?

Reply:
It looks normal now, I thought I wasn't seeing `Output of` in bold.

Reply:
Fixed with CSS :)